README.md

```diff
@@ -173,3 +173,98 @@ rails driver:hmis:dump_graphql_schema
 ```
 
 To pick up the local schema changes on the frontend, run the `graphql:codegen` script (see above).
+
+## Project Structure
+
+This project follows the [bulletproof-react](https://github.com/alan2207/bulletproof-react/blob/master/docs/project-structure.md) structure.
+
+```
+src
+|
++-- api               # graphql queries/mutations/fragments
+|
++-- app               # application layer containing:
+|   +-- App.tsx              # main application component
+|   +-- AppProvider.tsx      # application provider that wraps the entire application
+|   +-- protectedRoutes.ts   # define routes for the authorized applciation
++-- assets            # assets folder can contain all the static files such as images, fonts, etc.
+|
++-- components        # shared components used across the entire application
+|
++-- config            # global configurations, exported env variables etc.
+|
++-- modules           # feature based modules
+|
++-- hooks             # shared hooks used across the entire application
+|
++-- providers         # shared contexts and providers? (added)
+|
++-- test              # test utilities and mocks
+|
++-- types             # shared types used across the application
+|
++-- utils             # shared utility functions
+```
+
+Important modules
+```
+app
+components             # no components, just a dir
+components/elements    # common components
+components/input       # base input components that are used across the app
+components/layout      # layout components that are used across the app
+components/table       # table components that are used across the app (could be a module or just in elements/ ?)
+components/pages       # ***keep or move to modules or src/app/pages?? these are not re-used
```

Comment:
aside from `NotFound` I agree that these should be moved out of components into modules.

I also wonder if we should update to use Bulletproof React's naming, i.e. `features` instead of `modules`. I'm not sure, but it might help us remember/enforce that everything going in top-level `components` is meant to be reusable across the app